lib/elastic_apm.rb

```diff
@@ -381,5 +391,29 @@ def add_filter(key, callback = nil, &block)
 
       agent&.add_filter(key, block || callback)
     end
+
+    def set_transaction_outcome(result: nil, http_status_code: nil)
+      return unless transaction = current_transaction
+      return transaction.outcome = result if result
+      return unless http_status_code
+      code = http_status_code.to_i
+      if code < 500
+        transaction.outcome = 'success'
+      elsif code >= 500
```

Comment:
It's never _neither_ above _and_ below 500, so this will do:

```ruby
transaction.outcome = http_status_code.to_i < 500 ? 'success' : 'failure'
```

Or if you prefer:

```ruby
transaction.outcome =
  if http_status_code.to_i < 500
    'success'
  else 
    'failure'
  end
```